### PR TITLE
feat: cfd 515 myfares UI tests

### PR DESCRIPTION
**Pull request by matt-twomey**
# Description

-   Adds UI tests for MyFares, makes LNUD the test user for UI tests, limits otherProducts to 100 in test, adds an error log for when a non-test user has >50 other products

# Testing instructions

-   Run UI tests
-  Ensure otherProducts are limited to 100 in test (I tested this locally)
-  Ensure an alert is triggered when a non-test user with more than 50 products loads the otherProducts page

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    clickElementByText,
+    clickRandomElementsInTable,
+    getHomePage,
+} from '../../support/helpers';
+
+describe('Able to access services in my fares', () => {
+    it('Able to access service pages', () => {
+        getHomePage();
+        clickElementById('manage-fares-link');
+        clickRandomElementsInTable('govuk-table__body', 'service-link');
```
Comment (RobCatton) [inline repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts:13]: Do we need an assert inside here or inside the function?

Comment (RobCatton) [reply repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts:13]: And inside the other tests, if so

File: repos/fdbt-site/cypress_tests/cypress/support/helpers.ts [hunk 2]
```diff
@@ -462,6 +468,20 @@ export const completeMultipleProducts = (
...
 };
 
+export const clickRandomElementsInTable = (tableName: string, elementId: string): void => {
+    getElementByClass(tableName)
+        .find('tr')
+        .then((elm) => {
+            const numberOfServices = elm.length;
+            for (let i = 0; i < 3; i++) {
+                const randomSelector = getRandomNumber(0, numberOfServices - 1);
+                elm.get(randomSelector).setAttribute('selected', 'selected');
```
Comment (RobCatton) [inline repos/fdbt-site/cypress_tests/cypress/support/helpers.ts:478]: What are we seeking to achieve by clicking on the services?

File: repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        clickElementByText('Services');
+        clickRandomElementsInTable('govuk-table__body', 'service-link');
+    });
+});
+
+describe('Able to access other products in my fares', () => {
+    before(() => {
+        addFlatFareProductIfNotPresent();
+    });
+    it('Able to access other products pages', () => {
```
Comment (RobCatton) [inline repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts:27]: Could we change the name of the test to flow on from the word 'it' ?

File: repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts (hunk 1 above)
Comment (matt-twomey) [reply repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts:13]: So by clicking through the pages you're asserting that that journey works - if any elements aren't present/redirection doesn't occur correctly, it'll fail. That sound alright?

File: repos/fdbt-site/cypress_tests/cypress/support/helpers.ts (hunk 2 above)
Comment (matt-twomey) [reply repos/fdbt-site/cypress_tests/cypress/support/helpers.ts]: So this asserts that we can navigate to the pointToPointProducts page for a random set of services, and to the productDetails page for a random set of 'other products', depending on what's passed to it. I have removed line 478 though, as I don't think it was doing anything!

File: repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts (hunk 3 above)
Comment (matt-twomey) [reply repos/fdbt-site/cypress_tests/cypress/integration/myFares/myFares.ts]: Good shout, done that 👍

**Review by jacsamell — APPROVED** [review]

**Pull request merged**